### PR TITLE
Fix connector

### PR DESCRIPTION
**Pull request by asvetlov**
Prevent hang on connection acquiring after connection failure

### CODE REVIEW

File: tests/test_connector.py [hunk 1]
```diff
@@ -343,6 +343,28 @@ def test_release_close(loop):
     assert proto.close.called
 
 
+def test__drop_acquire_per_host1(loop):
```
Comment (webknjaz) [inline tests/test_connector.py:346]: @asvetlov is this `double__underscore` notation intentional?

Comment (asvetlov) [reply tests/test_connector.py:346]: Yes. It is a test for `_drop_acquire_per_host` method.

**Comment by pfreixes** [comment]
Why the clients that are waiting for an available connection hang in case of the previous one had an error? I cant see this bad behavior in the current code. Could you please point me to the right path?

Also, I don't get the point of the  `test_error_in_performing_request` test, how is it testing that situation?

**Comment by asvetlov** [comment]
self._waiters was not cleaned up, https://github.com/aio-libs/aiohttp/blob/2.3/aiohttp/connector.py#L357 said that `available` is not positive.
`connect()` method created a waiter but no other connection was exist, nobody set resumed the waiter future.

That's why `test_error_in_performing_request` failed.

I need review the logic for `available`, looks like `len(self._waiters)` is a wrong check.

**Comment by asvetlov** [comment]
Dropped a check for `len(self._waiters)`, `self._acquired` is enough.

**Comment by pfreixes** [comment]
Oks, got it. LGTM can be merged once all checks passes

**Comment by pfreixes** [comment]
IMHO this kind of bug is enough important to have a bugfix release for the 2.X series

**Comment by asvetlov** [comment]
@pfreixes will publish 2.3.4 just after merging

**Pull request merged**